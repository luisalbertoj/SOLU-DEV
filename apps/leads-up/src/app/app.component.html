<input type="text" [(ngModel)]="message.text" name="message.text" />
<button (click)="sendMessage()">send</button>
<a routerLink="/">Home</a>
<a routerLink="/auth/login">Login</a>
<a routerLink="/auth/register">Register</a>

<router-outlet></router-outlet>
