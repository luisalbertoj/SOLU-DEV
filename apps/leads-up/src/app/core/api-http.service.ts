import { Injectable } from '@angular/core';
// Atento a esta línea, deberás hacer que empate con tu environment
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { environment } from '../../environments/environment';

import { Observable } from 'rxjs';

const API_URL = environment.apiRoute;

@Injectable({
  providedIn: 'root',
})
export class ApiHttpService {
  protected headers;

  protected apiUrl = API_URL;

  constructor(private http: HttpClient) {
    this.headers = {
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        Accept: 'application/json',
      }),
    };
  }

  protected get(url: string): Observable<unknown> {
    return this.http.get(this.apiUrl + url, this.headers);
  }

  protected post({ url, payload }: { url: string; payload: unknown }) {
    return this.http.post('api/' + url, payload, this.headers);
  }
}
