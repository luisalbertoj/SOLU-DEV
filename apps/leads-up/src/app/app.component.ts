import { Component } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Message } from '@solu-dev/api-interfaces';

@Component({
  selector: 'solu-dev-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css'],
})
export class AppComponent {
  hello$ = this.http.get<Message>('/api/hello');
  constructor(private http: HttpClient) {}
}
