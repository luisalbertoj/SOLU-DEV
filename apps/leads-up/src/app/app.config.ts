import { Injectable } from '@angular/core';
import { JwtHelperService } from '@auth0/angular-jwt';

const helper = new JwtHelperService();

@Injectable()
export class AppConfig {
  load() {
    return new Promise((resolve) => {
      const token =
        localStorage.getItem('auth_token') === null
          ? false
          : localStorage.getItem('auth_token');
      if (token) {
        try {
          if (helper.isTokenExpired(token) === true) {
            localStorage.removeItem('auth_token');
          }
        } catch (exception) {
          localStorage.removeItem('auth_token');
        }
      }
      resolve(true);
    });
  }
}
