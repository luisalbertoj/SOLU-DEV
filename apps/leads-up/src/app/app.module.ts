import { HashLocationStrategy, LocationStrategy } from '@angular/common';
import { HttpClientModule, HTTP_INTERCEPTORS } from '@angular/common/http';
import { APP_INITIALIZER, NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { TitleStrategy } from '@angular/router';
import { NgxsReduxDevtoolsPluginModule } from '@ngxs/devtools-plugin';
import { NgxsLoggerPluginModule } from '@ngxs/logger-plugin';
import { NgxsModule } from '@ngxs/store';
import { MessageService } from 'primeng/api';
import { environment } from '../environments/environment';
import { AppComponent } from './app.component';
import { AppConfig } from './app.config';
import { AppRoutingModule } from './app.routing';
import { AccessInterceptor } from './core/interceptors/access.interceptor.service';
import { AuthInterceptor } from './core/interceptors/auth.interceptor.service';
import { PageTitleStrategy } from './core/services/page-title.service';
import { TokenProvider } from './core/services/token.service';
import { ToastService } from './shared/toast/services/toast.service';
import { ToastComponent } from './shared/toast/toast.component';
import { MessagesState } from './store/chat/chat.state';

export function servicesOnRun(config: AppConfig, token: TokenProvider) {
  return () => config.load().then(() => token.load());
}
@NgModule({
  declarations: [AppComponent],
  imports: [
    BrowserModule,
    BrowserAnimationsModule,
    AppRoutingModule,
    HttpClientModule,
    NgxsModule.forRoot([], {
      developmentMode: !environment.production,
    }),
    NgxsReduxDevtoolsPluginModule.forRoot({
      disabled: environment.production,
    }),
    NgxsLoggerPluginModule.forRoot({ disabled: environment.production }),
    NgxsModule.forRoot([MessagesState]),
    ToastComponent,
  ],
  providers: [
    AppConfig,
    TokenProvider,
    {
      provide: APP_INITIALIZER,
      useFactory: servicesOnRun,
      multi: true,
      deps: [AppConfig, TokenProvider],
    },
    { provide: HTTP_INTERCEPTORS, useClass: AuthInterceptor, multi: true },
    { provide: HTTP_INTERCEPTORS, useClass: AccessInterceptor, multi: true },
    {
      provide: TitleStrategy,
      useClass: PageTitleStrategy,
    },
    {provide: LocationStrategy, useClass: HashLocationStrategy},
    ToastService,
    MessageService,
  ],
  bootstrap: [AppComponent],
})
export class AppModule {}
