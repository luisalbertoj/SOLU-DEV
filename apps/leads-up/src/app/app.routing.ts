import { NgModule } from '@angular/core';
import { RouterModule, Routes } from '@angular/router';

const routes: Routes = [
  {
    path: 'auth',
    title: 'auth',
    loadChildren: () => import('./auth/auth.module').then((m) => m.AuthModule),
  },
  { path: '**', redirectTo: '/auth', pathMatch: 'full' },
];

@NgModule({
  imports: [
    RouterModule.forRoot(routes, { scrollPositionRestoration: 'enabled' }),
  ],
  exports: [RouterModule],
})
export class AppRoutingModule {}
