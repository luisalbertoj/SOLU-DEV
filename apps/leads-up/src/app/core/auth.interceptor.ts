import {
  HttpEvent,
  HttpHandler,
  HttpInterceptor,
  HttpRequest,
} from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable()
export class AuthInterceptor implements HttpInterceptor {
  intercept(
    req: HttpRequest<unknown>,
    next: HttpHandler
  ): Observable<HttpEvent<unknown>> {
    const token = localStorage.getItem('auth_token');

    if (!token) {
      return next.handle(req);
    }

    const request = req.clone({
      headers: req.headers.set('Authorization', `Bearer ${token}`),
    });

    return next.handle(request);
  }
}
