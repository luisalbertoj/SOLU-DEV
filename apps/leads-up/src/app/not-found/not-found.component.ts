import { CommonModule } from '@angular/common';
import { ChangeDetectionStrategy, Component } from '@angular/core';
import { RouterModule } from '@angular/router';
import { ButtonModule } from 'primeng/button';
import { AppComponent } from '../app.component';

@Component({
  selector: 'solu-dev-not-found',
  standalone: true,
  imports: [CommonModule, RouterModule, ButtonModule],
  templateUrl: './not-found.component.html',
  styleUrls: ['./not-found.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class NotFoundComponent {
  constructor(public app: AppComponent) {}
}
