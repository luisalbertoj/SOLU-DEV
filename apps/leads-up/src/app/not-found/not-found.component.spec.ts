import { ComponentFixture, TestBed } from '@angular/core/testing';
import { RouterTestingModule } from '@angular/router/testing';
import { AppComponent } from '../app.component';

import { NotFoundComponent } from './not-found.component';

describe('NotFoundComponent', () => {
  let component: NotFoundComponent;
  let fixture: ComponentFixture<NotFoundComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      declarations: [],
      imports: [RouterTestingModule],
      providers: [AppComponent],
    }).compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(NotFoundComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('this should be equal to the snapshot', () => {
    expect(compiled.innerHTML).toMatchSnapshot();
  });
});
