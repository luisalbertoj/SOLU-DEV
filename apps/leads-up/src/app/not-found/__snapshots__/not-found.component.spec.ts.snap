// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent this should be equal to the snapshot 1`] = `"<div class=\\"exception-body notfound\\"><div class=\\"exception-panel\\"><h1>404</h1><h3>not found</h3><p>The page that you are looking for does not exist</p><button type=\\"button\\" pbutton=\\"\\" label=\\"Go back to home\\" class=\\"p-element p-button p-component\\" tabindex=\\"0\\" ng-reflect-label=\\"Go back to home\\" ng-reflect-router-link=\\"/\\"><span class=\\"p-button-label\\">Go back to home</span></button></div><div class=\\"exception-footer\\"><img class=\\"exception-logo\\" src=\\"assets/layout/images/logo-light.png\\"><img class=\\"exception-appname\\" src=\\"assets/layout/images/appname-light.png\\"></div></div>"`;
